### PR TITLE
Using the flux2-sync helm chart in the examples.

### PR DESCRIPTION
**Pull request by swade1987**

### CODE REVIEW

**Comment by swade1987** [comment]
@stefanprodan could you please set the required tests to be the following now moving forward please 

- `tests / e2e-helm-install-with-github`
- `tests / e2e-flux-bootstrap-with-github`

**Comment by swade1987** [comment]
> @swade1987 e2e tests can't be required as they don't run on forks.

@stefanprodan I meant for the main branch.

File: examples/helm-install/main.tf [hunk 1]
```diff
@@ -18,55 +18,141 @@ terraform {
...
-  create_namespace = true
+  name      = "flux2"
+  namespace = "flux-system"
 
   # Allows us to upgrade the version of flux from another repo and terraform not roll it back.
   lifecycle {
     ignore_changes = [
       version
     ]
   }
```
Comment (stefanprodan) [inline examples/helm-install/main.tf:100]: I don't think this is Ok, by doing this how will people upgrade with TF is the chart version is ignored?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -160,3 +160,71 @@ jobs:
             https://api.github.com/repos/fluxcd-testing/${{ steps.vars.outputs.test_repo_name }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITPROVIDER_BOT_TOKEN }}
+  e2e-helm-install-with-github:
+    runs-on: ubuntu-latest
```
Comment (stefanprodan) [inline .github/workflows/tests.yaml:164]:
```suggestion
    runs-on: ubuntu-latest
    needs: build
```

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -160,3 +160,71 @@ jobs:
...
+      - name: Setup Flux CLI
+        uses: fluxcd/flux2/action@534684601ec8888beb0cc4f51117b59e97606c4d # v2.2.3
+      - name: Set outputs
+        id: vars
+        run: |
+          REPOSITORY_NAME=${{ github.event.repository.name }}
+          BRANCH_NAME=${GITHUB_REF##*/}
+          COMMIT_SHA=$(git rev-parse HEAD)
+          PSEUDO_RAND_SUFFIX=$(echo "${BRANCH_NAME}-${COMMIT_SHA}" | shasum | awk '{print $1}')
+          TEST_REPO_NAME="${REPOSITORY_NAME}-${PSEUDO_RAND_SUFFIX}"
```
Comment (stefanprodan) [inline .github/workflows/tests.yaml:186]: This will generate the same repo name as the for the bootstrap test and will fail.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @swade1987 🏅

**Pull request merged**